src/Main.java

```diff
@@ -0,0 +1,172 @@
+
+
+//Класс Main реализует «Телефонный справочник» на основе контейнера HashMap<String, String>
+//в качестве ключа используется номер телефона в виде строки, а в качестве значения - фамилия
+//класс включает в себя метод main который запускает программу, а также включены методы:
+//add - добавляет запись по заданным номеру телефона и фамилии
+//del - удаляет запись по номеру телефона
+//save - сохраняет в текстовом файле phone.txt
+//load - загружает из текстового файла phone.txt
+//PrintPhonebook - выводит на екран все записи
+//FindSurname - производит поиск фамилии по номеру телефона
+//FindNumberPhone - производит поиск списка номеров по фамилии
+
+public class Main  {
+    private static Map<String, String> string = new HashMap<String, String>();
+
+    //addPB - добавляет запись по заданным номеру телефона и фамилии
+    private static void add(String phone, String name) {
+        string.put(phone, name);
+    }
+
+    //delPB - удаляет запись по номеру телефона
+    private static void del(String phone) {
+        string.remove(phone);
+    }
+
+    //savePB - сохраняет БД в текстовом файле phone.txt
+    private static void save() throws IOException {
+        BufferedWriter writer = new BufferedWriter(new FileWriter(new File("phone.txt")));
+        for(Map.Entry<String,String> k: string.entrySet()){
+            writer.write(k.getKey() + " " + k.getValue()+System.lineSeparator());
+        }
+        writer.close();
+    }
+
+    //loadPB - загружает БД из текстового файла phone.txt
+    //производит проверку на наличие файла
+    public static void load() throws IOException {
+        File file = new File("phone.txt");
+        if (file.exists()){
+            BufferedReader reader = new BufferedReader(new FileReader(new File("phone.txt")));
+            String act;
+            while ((act=reader.readLine())!=null) {
+                String[] dat = act.split(" ");
+                string.put(dat[0], dat[1]);
+            }
+            reader.close();
+        }
+    }
+
+    //PrintPhonebook - выводит на екран все записи БД
+    public static void PrintPhonebook(){
+        System.out.println("Телефонный справочник: ");
+        for(Map.Entry<String,String> k: string.entrySet()){
+            System.out.println(k.getValue()+": "+ k.getKey());
+        }
+    }
+
+    //FindSurname - производит поиск фамилии по номеру телефона заданому в качестве аргумента
+    //возвращает строку
+    public static String FindSurname(String number){
+        String result = string.get(number);
+        if (result == null) return "абонент с таким номером не найдей";
+        return result;
+    }
+
+    //FindNumberPhone - производит поиск списка номеров по фамилии заданой в качестве аргумента
+    //возвращает массив строк
+    public static String[] FindNumberPhone(String surname){
+        List <String> result = new ArrayList<String>();
+        for (Map.Entry entry : string.entrySet()) {
```

Comment:
Use parameterized entry, except casting entry key and value:
  ```suggestion
            for (Map.Entry<String, String> entry : string.entrySet()) {
                if (surname.equalsIgnoreCase(entry.getValue())) {
                    result.add(entry.getKey());
                }
        }
```